### PR TITLE
use pytest to discover and run all unittests

### PR DESCRIPTION
**Pull request by fa-me**
The usage of `pytest` renders manual appending of tests unnecessary and gives a cleaner output. Doctests are also detected and run. This PR removes the old `unittests.py` which had the same purpose.

The commits 59baddf and 4c2483d should demonstrate correct detection of failing tests.

### CODE REVIEW

File: dodo.py [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
 import glob
 import os.path
 
+
 def task_test():
-    def integration_test_action():
-        # printing to stderr so that doit does not capture the output
-        print("### Integration tests are not yet implemented ###", file=sys.stderr)
+    return {
+        'actions': ['pytest'],
```
Comment (Sh4pe) [inline dodo.py]: Why no `--cov=./ --doctest-modules` here? When the recipes in `appveyor.yml` and `.travis.yml` could remain `doit test`.

File: requirements.txt [hunk 2]
```diff
@@ -7,4 +7,9 @@ dash-html-components
 
 # The latest version of doit supporting Python 2 is 0.29.0
 doit==0.29.0; python_version < '3.0'
-doit; python_version >= '3.0'
\ No newline at end of file
+doit; python_version >= '3.0'
+
+pytest==4.0.0; python_version < '3.0'
```
Comment (Sh4pe) [inline requirements.txt:16]: Why is version pinning necessary for `pytest`?

File: pynance/dummy.py [hunk 3]
```diff
@@ -3,11 +3,12 @@
 This is just a placeholder.
 """
 
+
```
Comment (Sh4pe) [inline pynance/dummy.py]: Ok, I see that I need to setup this autoformatting as well :-D

**Review by Sh4pe — CHANGES_REQUESTED** [review]

**Comment by Sh4pe** [comment]
Maybe we can use this PR to remove `dummy.py` and `dummy_test.py` as well, what do you think?

File: dodo.py (hunk 1 above)
Comment (fa-me) [reply dodo.py]: yes, you're right

File: requirements.txt (hunk 2 above)
Comment (fa-me) [reply requirements.txt:16]: sadly yes, in order to work on all machines 😞 
it seems pytest and pytest-cov need to have matching versions, and the osx machine did not pull the right ones. Only earlier versions work well with python 2

**Comment by fa-me** [comment]
This fixes #41

**Comment by fa-me** [comment]
codecov fails because `dodo.py` is untested.
@Sh4pe should we remove dodo from coverage scope, should we test it (how?) or should we drop the 90% target for a patch? In my eyes 90% coverage for the total project is enough as a requirement.

File: requirements.txt (hunk 2 above)
Comment (Sh4pe) [reply requirements.txt:16]: Ok. Then let's leave it like this.

File: dodo.py (hunk 1 above)
Comment (Sh4pe) [reply dodo.py]: Cool :)

**Comment by Sh4pe** [comment]
> codecov fails because `dodo.py` is untested.
> @Sh4pe should we remove dodo from coverage scope, should we test it (how?) or should we drop the 90% target for a patch? In my eyes 90% coverage for the total project is enough as a requirement.

I think we should remove `dodo.py` from the coverage scope and drop the 90% target for the patch. It is absolutely okay (very good even) if the total project has 90% coverage IMO as well.

**Comment by fa-me** [comment]
@Sh4pe I removed the 90% coverage of the patch. I think there is no need to manually exclude the dodo.py - it doesn't contain much code and will barely affect the total coverage. In fact we might think of testing this content as well, should not be too hard.

**Comment by Sh4pe** [comment]
Awesome! Will merge this now!

**Pull request merged**